presto-main/src/main/java/com/facebook/presto/operator/project/PageProcessor.java

```diff
@@ -84,17 +86,17 @@ public PageProcessorOutput process(ConnectorSession session, Page page)
             }
 
             if (projections.isEmpty()) {
-                return new PageProcessorOutput(page.getRetainedSizeInBytes(), Iterators.singletonIterator(new Page(selectedPositions.size())));
+                return new PageProcessorOutput(() -> calculatePageRetainedSizeWithoutLoading(page), singletonIterator(new Page(selectedPositions.size())));
```

Comment:
The "Page" in this method name is redundant since it takes a page. It could be `calculateRetainedSizeWithoutLoading(page)`.